python/mlc_chat/quantization/ft_quantization.py

```diff
@@ -117,11 +122,15 @@ def visit_module(self, name: str, node: nn.Module) -> Any:
                 ret_node: Any
                     The new node to replace current node.
                 """
-                if isinstance(node, nn.Linear) and node.out_features != "vocab_size":
-                    # This skips lm_head in llama; otherwise the quantized shape would be an op
-                    # See https://github.com/mlc-ai/mlc-llm/issues/1723
+                if isinstance(node, nn.Linear):
```

Comment:
it's safer to check the shape and decide if it can be quantized with ft, previous we did https://github.com/mlc-ai/mlc-llm/blob/main/mlc_llm/quantization/ft_quantization.py#L207

Reply:
Thanks for the review! Updated to:
```python
if (
    is_final_fc(name)
    or node.weight.dtype == "float32"
    or node.out_features % DataType(self.config.storage_dtype).bits != 0
):
  # Fall back to GroupQuantize
```

I'm guessing the `% 8` in the original corresponds to `storage_dtype` right? Let me know if this is what you have in mind!

Reply:
https://github.com/tlc-pack/cutlass_fpA_intB_gemm/blob/main/cutlass_kernels/cutlass_preprocessors.cc#L436, it's four bytes (int4 x 8 or int8 x 4)

Reply:
I see; are the `int4` and `int8` `storage_dtype` or `quantize_dtype`?

Reply:
they are quantize types

Reply:
Just made the change